### ISSUE TITLE
RightCurlyCheck: false positive when do while brace is after method call

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#3364]
````
$ cat TestClass.java
public class TestClass {
    void method() {
        do {
        } while (false);
        do {
            "".substring(0);
        } while (false);
        do {
            if (false)
                return;
        } while (false);
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="RightCurly">
<property name="option" value="alone" />
<property name="tokens" value="LITERAL_WHILE, LITERAL_DO" />
</module>
    </module>
</module>

$ java -jar checkstyle-7.0-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:7:9: '}' at column 9 should be alone on a line. [RightCurly]
[ERROR] TestClass.java:11:9: '}' at column 9 should be alone on a line. [RightCurly]
Audit done.
Checkstyle ends with 2 errors.
````

I am expecting similar validation between line 4, 7, and 9 for the brace position. The contents inside the braces shouldn't matter for this check, only where the braces are located.
Since line 4 is not producing a violation, line 7 and 9 shouldn't produce a violation either I would think.

**Comment by MEZk** [comment]
@rnveach 
No, line 7 and 11 should produce violations as you set brace policy to "ALONE". 
According to the check's logic if block is empty the check does not produce violation, but I agree that such behaviour should be documented or changed.

**Comment by rnveach** [comment]
@MEZk 
I see now that it puts a violation whenever it is not empty.

http://checkstyle.sourceforge.net/property_types.html#rcurly
`same` describes the options as relative to `the next part of a multi-block statement`. `DO-WHILE` has no next multi-block statement, so to me it is technically alone which is why I thought that was right option.
It seems it really refers to if any text, not just multi-blocks, is allowed after the brace on the same or next line.

You are right, it doesn't look like the `empty` block scenario is not mentioned anywhere in the documentations. It was added with #213.

I will close this and try to re-tune my configuration for right and left curlies.

**Comment by MEZk** [comment]
@rnveach 
>the next part of a multi-block statement

That description was taken from Guava's code style guide and now it tends to stale, I guess.
